src/pages/resultsView/mutation/Mutations.tsx

```diff
@@ -173,13 +121,96 @@ export default class Mutations extends React.Component<
     }
 
     protected handleTabChange(id: string) {
-        // update the hash if routing exits
-        if (this.props.routing) {
-            this.props.routing.updateRoute({ mutationsGeneTab: id });
-        }
-        // update the observable if no routing
-        else {
-            this.mutationsGeneTab = id;
+        this.props.store.setSelectedGeneSymbol(id);
+    }
+
+    @computed get geneTabContent() {
+        if (
+            this.props.store.selectedGene &&
+            this.props.store.mutationMapperStoreForSelectedGene
+        ) {
```

Comment:
Assuming that we lazy init the mutation mapper stores within `getMutationMapperStore`, we can do something like this:
```suggestion
        if (this.selectedGene) {
            const mutationMapperStore = this.props.store.getMutationMapperStore(this.selectedGene)
```